### DISCUSSION TITLE
Create Langeuage Server to edit tid in VSCode, or provide highlight in codemirror inside wiki

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#7248] [Ideas]
1. Use https://langium.org/ to create a Language server
2. Write a server side plugin to run this server, provide a new server router that provides LSP
3. Codemirror or other editor can use this as syntax highlighter
4. Release this as a new VSCode plugin

**Comment by Jermolene** [comment]
Hi @linonetwo it would be great to have a TiddlyWiki language server. I think that on the TW side we still need to update some parse rules to set start/end ranges in their output.
